lib/fibonacci.rb

```diff
@@ -1,8 +1,17 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
-# Hint, you may want a recursive helper method
+# Time Complexity - O(n)
+# Space Complexity - O(n)  
+
 def fibonacci(n)
```

Comment:
This works, but you're storing all fibonacci numbers in the sequence, but you only need to remember the last two numbers.  Can you find a way to reduce the amount of memory used?